modules/webgl/src/adapter/resources/webgl-command-buffer.ts

```diff
@@ -171,9 +173,12 @@ function _copyTextureToBuffer(device: WebGLDevice, options: CopyTextureToBufferO
     const sourceWidth = width || framebuffer.width;
     const sourceHeight = height || framebuffer.height;
 
-    // TODO - hack - should be deduced
-    const sourceFormat = GL.RGBA;
-    const sourceType = GL.UNSIGNED_BYTE;
+    const format = TEXTURE_FORMATS[framebuffer.texture.format];
```

Comment:
Nit: Would it feel better to have some function to query attributes for a format rather than indexing such a messy table directly?

Reply:
Just found `getWebGLTextureParameters`, that works great!